torch/csrc/autograd/record_function.h

```diff
@@ -103,6 +103,22 @@ struct TORCH_API RecordFunction {
 
   void end();
 
+  inline void setThreadId(uint16_t threadId) const {
+    threadId_ = threadId;
```

Comment:
Can we have only one variable `threadId_` which is int32_t instead of having overrideThreadId_ and threadId_. By default threadId_ would be set to -1. If we do need to record the thread id for a function, we set threadId_ to the appropriate value. The boolean check for whether or not overrideThreadId is set can be done by just checking whether threadId_ is -1 or not?

Reply:
" If we do need to record the thread id for a function, we set threadId_ to the appropriate value"

I'm not sure how we'd implement this - the current approach always sets the thread_id on the `RecordFunction`, but it is left to the user using the `RecordFunction` to determine if they actually want to use this or not using the boolean. 

Basically, I"m not sure how we'd figure out if we do need to record the thread ID for a function or not in the profiler's callbacks or not, without hardcoding a list of functions or something like that. We can modify the callbacks in `profiler::enableProfiler()` to return the `thread_id`, but that would be a significant refactor since we'd have to pass the return value through several layers of callback managers as well.

Reply:
Synced offline.